commit b1b8f5d89d06a994773e22ad4b7fe1036b37f8ab
Author: Stefan Liebler <stli@linux.vnet.ibm.com>
Date:   Thu Apr 14 12:21:53 2016 +0200

    Fix strfmon_l: Use specified locale for number formatting [BZ #19633]
    
    The commit 985fc132f23dbb83de76c5af9e783ef1b5900148
    "strfmon_l: Use specified locale for number formatting [BZ #19633]"
    introduced an elf/check-abi-libc testfailure due to __printf_fp_l
    on architectures which use sysdeps/ieee754/ldbl-opt/math_ldbl_opt.h.
    
    This patch uses libc_hidden_def instead of ldbl_hidden_def.
    The ldbl_strong_alias is removed due to the rename of ___printf_fp_l
    to __printf_fp_l.
    
    ChangeLog:
    
            * stdio-common/printf_fp.c (__printf_fp_l):
            Rename ___printf_fp_l to __printf_fp_l and
            remove strong alias. Use libc_hidden_def instead
            of ldbl_hidden_def macro.

Index: b/stdio-common/printf_fp.c
===================================================================
--- a/stdio-common/printf_fp.c
+++ b/stdio-common/printf_fp.c
@@ -153,9 +153,9 @@ static wchar_t *group_number (wchar_t *b
 
 
 int
-___printf_fp_l (FILE *fp, locale_t loc,
-		const struct printf_info *info,
-		const void *const *args)
+__printf_fp_l (FILE *fp, locale_t loc,
+	       const struct printf_info *info,
+	       const void *const *args)
 {
   /* The floating-point value to output.  */
   union
@@ -1247,14 +1247,13 @@ ___printf_fp_l (FILE *fp, locale_t loc,
   }
   return done;
 }
-ldbl_hidden_def (___printf_fp_l, __printf_fp_l)
-ldbl_strong_alias (___printf_fp_l, __printf_fp_l)
+libc_hidden_def (__printf_fp_l)
 
 int
 ___printf_fp (FILE *fp, const struct printf_info *info,
 	      const void *const *args)
 {
-  return ___printf_fp_l (fp, _NL_CURRENT_LOCALE, info, args);
+  return __printf_fp_l (fp, _NL_CURRENT_LOCALE, info, args);
 }
 ldbl_hidden_def (___printf_fp, __printf_fp)
 ldbl_strong_alias (___printf_fp, __printf_fp)
